src/libraries/System.Private.CoreLib/src/System/SpanHelpers.T.cs

```diff
@@ -1672,6 +1672,15 @@ internal static int IndexOfAnyChar(ref char searchSpace, char value0, char value
         {
             if (PackedSpanHelpers.PackedIndexOfIsSupported && typeof(TValue) == typeof(short) && PackedSpanHelpers.CanUsePackedIndexOf(value0) && PackedSpanHelpers.CanUsePackedIndexOf(value1))
             {
+                if ((*(char*)&value0 ^ *(char*)&value1) == 0x20)
+                {
+                    char lowerCase = (char)Math.Max(*(char*)&value0, *(char*)&value1);
```

Comment:
I assume this extra overhead is worth it even for short inputs or inputs where the target is found early?

Reply:
If the arguments to `IndexOfAny` are constants, the JIT makes this whole thing "free" for us, and you'll just end up with a call into the appropriate helper.

If the arguments are not constant, you will see a small regression for early matches.
Ideally, this would all be hidden behind `RuntimeHelpers.IsKnownConstant`, but I didn't have any luck in getting it to work right (it always reports the values as non-const, presumably because they've been address exposed, cc: @EgorBo).